src/coredump/coredump.c

```diff
@@ -335,15 +376,11 @@ static int save_external_coredump(
         if (r < 0)
                 return log_error_errno(r, "Failed to determine coredump file name: %m");
 
-        r = tempfn_random(fn, NULL, &tmp);
-        if (r < 0)
-                return log_error_errno(r, "Failed to determine temporary file name: %m");
-
         mkdir_p_label("/var/lib/systemd/coredump", 0755);
```

Comment:
yeah, this all is a bit chaotic... we often use the mkdir_label() where we shouldn't and mkdir() where we probably should... One of those days we should probably clean this up...